packages/nft-erc1155-provider/lib/NftErc1155Provider.ts

```diff
@@ -0,0 +1,126 @@
+import { Provider } from '@liquality/provider'
+import { addressToString } from '@liquality/utils'
+import { ensure0x, normalizeTransactionObject } from '@liquality/ethereum-utils'
+import { NftProvider, Address } from '@liquality/types'
+import { StandardError } from '@liquality/errors'
+
+import { Contract } from '@ethersproject/contracts'
+import { Signer } from '@ethersproject/abstract-signer'
+
+import NftErc1155_ABI from './NftErc1155_ABI.json'
+
+const erc1155InterfaceID = '0xd9b67a26'
+
+export default class NftErc1155Provider extends Provider implements Partial<NftProvider> {
+  _contract: Contract
+  _signer: Signer
+  _contractCache: { [address: string]: boolean }
+
+  constructor(signer: Signer) {
+    super()
+
+    this._signer = signer
+    this._contract = new Contract('0x0000000000000000000000000000000000000000', NftErc1155_ABI, this._signer)
+    this._contractCache = {}
+  }
+
+  async balance(contract: Address | string, tokenIDs: number | number[]) {
+    await this._attach(contract)
+
+    if (!tokenIDs) return 0
+
+    if (tokenIDs.constructor === Array) {
+      const addresses = Array.from({ length: tokenIDs.length }).fill(await this._signer.getAddress())
+      const amounts = await this._contract.functions.balanceOfBatch(addresses, tokenIDs)
+      return amounts[0].map((amount: any) => amount.toNumber())
+    }
+
+    const amount = await this._contract.functions.balanceOf(await this._signer.getAddress(), tokenIDs)
+    return amount[0].toNumber()
+  }
+
+  async transfer(
+    contract: Address | string,
+    receiver: Address | string,
+    tokenIDs: number | number[],
+    values: number[],
+    data = '0x00'
+  ) {
+    await this._attach(contract)
+
+    if (
+      tokenIDs &&
+      tokenIDs.constructor === Array &&
+      values &&
```

Comment:
You can combine the checks like this:
`isArray(values?.constructor)`

so `values && values.constructor === Array` becomes ``isArray(values?.constructor)``